src/lib/autocomplete/autocomplete-trigger.ts

```diff
@@ -343,6 +342,29 @@ export class MdAutocompleteTrigger implements ControlValueAccessor, OnDestroy {
     }
   }
 
+  /** Determines the index of the active option. */
+  private _getActiveOptionIndex(): number {
+    let optionIndex = this.autocomplete._keyManager.activeItemIndex || 0;
+
+    // If there are any option groups, we need to offset
+    // the index by the amount of groups that come before the option.
+    if (this.autocomplete.optionGroups.length) {
+      const options = this.autocomplete.options.toArray();
+      const groups = this.autocomplete.optionGroups.toArray();
+      let groupCounter = 0;
+
+      for (let i = 0; i < optionIndex + 1; i++) {
```

Comment:
This process looks pretty similar to how select calculates option index [here](https://github.com/crisbeto/material2/blob/8ad659f07b384afdd77cc3887c66e99141905f6d/src/lib/select/select.ts#L1048). Given this is a common operation for option groups, maybe we could move this into a reusable function in core/option? A bit DRY-er.